backend/src/Edna/Analysis/FourPL.hs

```diff
@@ -49,8 +63,49 @@ instance ToSchema Params4PL where
   declareNamedSchema Proxy =
     declareNamedSchema @Params4PLTuple Proxy
 
+type AnalysisResult = Either Text Params4PL
+
+instance Buildable (AnalysisResult) where
+  build = build . (coerce :: AnalysisResult -> Either Text Params4PL)
+
+instance Buildable (ForResponseLog AnalysisResult) where
+  build = buildForResponse
+
+data Params4PLReq = Params4PLReq
+  { plreqExperiment :: ExperimentId
+  , plreqData :: [(Double, Double)]
+  } deriving stock (Generic, Show, Eq)
+
+deriveJSON ednaAesonWebOptions ''Params4PLReq
+
+data Params4PLResp = Params4PLResp
+  { plrspExperiment :: ExperimentId
+  , plrspData :: AnalysisResult
+  } deriving stock (Generic, Show, Eq)
+
+deriveJSON ednaAesonWebOptions ''Params4PLResp
+
+callPythonAnalysis :: String -> Edna (Either String [Params4PLResp])
+callPythonAnalysis request = do
+  logDebug $ toText $ "Python data request: " <> request
+  analysisDir <- liftIO $ fromMaybe (".." </> "analysis") <$> (lookupEnv "ANALYSIS_DIR")
```

Comment:
```suggestion
  analysisDir <- liftIO $ fromMaybe (".." </> "analysis") <$> lookupEnv "EDNA_ANALYSIS_DIR"
```

Please rename `ANALYSIS_DIR` to `EDNA_ANALYSIS_DIR` here and in nix files to avoid possible name collisions. Also brackets look redundant.